### ISSUE TITLE
Kubectl 1.10.0 is unable to create deployments on clusters running 1.8.7 

### ISSUE THREAD

**Issue by ihoegen** [kubernetes/kubectl#385]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

/kind bug

**Kubernetes version** (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-27T00:13:02Z", GoVersion:"go1.9.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.7", GitCommit:"b30876a5539f09684ff9fde266fda10b37738c9c", GitTreeState:"clean", BuildDate:"2018-01-16T21:52:38Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: AWS
- **OS**: macOS 10.13.3
- **Kernel**: Darwin 17.4.0
- **Install tools**: brew

**What happened**:

When creating a sample deployment using the command `kubectl create -f deployment.yaml`, the following output appears:

`Error from server (NotAcceptable): unknown`

**What you expected to happen**:

When running the same command again, `kubectl create -f deployment.yaml`, with kubectl version 1.9.3, the following output appears:

`deployment "nginx-deployment" created`

A better error message describing incompatibility might be useful.

**How to reproduce it** (as minimally and precisely as possible):

1. Use the following config, and save it as `deployment.yaml`:

``` yaml
apiVersion: apps/v1beta2 
kind: Deployment
metadata:
  name: nginx-deployment
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 2
  template: 
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80
```

2. Run `kubectl create -f deployment.yaml` using kubectl 1.10.0

**Comment by ihoegen** [comment]
Opened issue here as well: https://github.com/kubernetes/kubernetes/issues/61943

**Comment by jheiss** [comment]
This can probably be closed since the other ticket has been fixed and closed.

**Comment by liggitt** [comment]
resolved in 1.10.1

**Comment by jonathanbeber** [comment]
Hi, 

I'm with client `1.11.0` and server `v1.8.8` and facing the same issue. Should I open a new issue?

**Comment by liggitt** [comment]
> I'm with client 1.11.0 and server v1.8.8 and facing the same issue. Should I open a new issue?

No. 1.8.x servers have a known bug that makes them respond incorrectly to requests containing Accept headers they don't recognize. This was fixed in 1.9.0. 

1.8.x is outside the supported skew window for kubectl 1.11.0
